app/models/miq_expression.rb

```diff
@@ -566,6 +566,10 @@ def self._to_ruby(exp, context_type, tz)
     clause
   end
 
+  def normalize_date_time(rel_time, tz, mode = "beginning")
```

Comment:
absolutely agree, this is the goal! The only concern is complexity of PR. I was planning to create multiple PRs  for task of changing method's scope. What is preferable way: multiple PRs or multiple commits on single PR ?